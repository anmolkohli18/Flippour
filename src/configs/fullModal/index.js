import { connect } from "react-redux";
import { FullModalComponent } from "./component";

const stateToProps = state => {
  const message = state.FullModalReducer.fullMessages;
  return {
    component: message.length > 0 ? message[0] : null
  };
};

export const FullModalLayer = connect(stateToProps)(FullModalComponent);
