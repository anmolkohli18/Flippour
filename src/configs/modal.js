import React, { Component } from "react";
import { View, Animated, Dimensions, Platform } from "react-native";
// import { SoundDecorator } from "../decorators/sound";
import { Text, Button } from "native-base";
import { connect } from "react-redux";
import { SettingsDecorator } from "../decorators/settings";
import { ModalActions } from "../actions/";

const styles = {
  ToastContainer: {
    position: "absolute",
    zIndex: 99,
    top: 0,
    width: Dimensions.get("window").width,
    alignItems: "stretch"
  },
  Background: {
    backgroundColor: "#c3524c",
    paddingTop: Platform.OS === "ios" ? 22 : 10,
    paddingBottom: 10
  },
  Text: {
    color: "white",
    textAlign: "center"
  }
};

// @SoundDecorator(["notification_in"])
@SettingsDecorator()
class ModalComponent extends Component {
  animation = {
    fadeAnim: new Animated.Value(0),
    positionAnim: new Animated.Value(-50)
  };

  state = {
    message: "No message has been entered",
    midTransition: false
  };

  start = () => {
    // this.props.sounds.notification_in.setCurrentTime(0.09);
    // this.props.sounds.notification_in.play();
    Animated.parallel([
      Animated.timing(this.animation.fadeAnim, {
        toValue: 1,
        duration: 150,
        useNativeDrier: true
      }),
      Animated.timing(this.animation.positionAnim, {
        toValue: 0,
        duration: 150,
        useNativeDrier: true
      })
    ]).start();
  };

  remove = () => {
    Animated.parallel([
      Animated.timing(this.animation.fadeAnim, {
        toValue: 0,
        duration: 150,
        useNativeDrier: true
      }),
      Animated.timing(this.animation.positionAnim, {
        toValue: -50,
        duration: 150,
        useNativeDrier: true
      })
    ]).start();
  };

  shouldComponentUpdate = () => {
    return !this.state.midTransition;
  };

  componentWillReceiveProps = nextprops => {
    if (nextprops.messages.length > 0 && !this.state.midTransition) {
      const message = nextprops.messages[nextprops.messages.length - 1];
      this.setState(
        () => {
          return { message: message.text, midTransition: true };
        },
        () => {
          this.start();
          setTimeout(() => {
            this.remove();
            this.props.removeMessage();
            this.setState(() => {
              return { midTransition: false };
            });
          }, message.time + 600);
        }
      );
    }
  };

  render = () => {
    const { modal, modalText } = this.props.settings.theme;

    const containerStyles = Object.assign({}, styles.ToastContainer, {
      opacity: this.animation.fadeAnim,
      transform: [{ translateY: this.animation.positionAnim }]
    });

    return (
      <Animated.View style={containerStyles}>
        <View style={Object.assign({}, styles.Background, modal)}>
          <Text style={Object.assign({}, styles.Text, modalText)}>
            {this.state.message}
          </Text>
        </View>
      </Animated.View>
    );
  };
}

const stateToProps = state => {
  return {
    messages: state.ModalReducer.messages
  };
};

const actionsToProps = dispatch => {
  return {
    removeMessage: () => dispatch(ModalActions.removeMessage())
  };
};

export const ModalLayer = connect(stateToProps, actionsToProps)(ModalComponent);
