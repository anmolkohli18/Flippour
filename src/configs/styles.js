import React, { PureComponent } from "react";
import { StatusBar } from "react-native";
import { connect } from "react-redux";
import { Container, StyleProvider, getTheme } from "native-base";
import variables from "../styles/theme";
import * as Colors from "../styles/variables";
import { SettingsDecorator } from "../decorators/settings";

@SettingsDecorator()
class StylesComponent extends PureComponent {
  render = () => {
    return (
      <StyleProvider style={getTheme(variables)}>
        <Container style={this.props.settings.theme.container}>
          <StatusBar
            backgroundColor={this.props.settings.theme.modal.backgroundColor}
            barStyle={this.props.theme === "LIGHT" ? "dark-content" : "light-content"}
          />
          {this.props.children}
        </Container>
      </StyleProvider>
    );
  }
}

export const StylesLayer = connect(state => {
  return {
    theme: state.SettingsReducer.theme
  };
})(StylesComponent);
