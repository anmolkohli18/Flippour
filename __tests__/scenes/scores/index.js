import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import { ReduxLayer } from "../../../src/configs/redux";
import { ScoresScene } from "../../../src/scenes/scores/index.js";

test("ScoresScene renders correctly", () => {
  const tree = renderer
    .create(
      <ReduxLayer>
        <ScoresScene
          scores={[{ score: 99, level: 24 }]}
          featured={{ score: 200, level: 100 }}
          title={"title here"}
          subtitle={"subtitle here"}
        />
      </ReduxLayer>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
