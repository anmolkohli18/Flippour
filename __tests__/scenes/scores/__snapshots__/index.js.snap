// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoresScene renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "zIndex": 9,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "left": 0,
            "minHeight": 42,
            "minWidth": 50,
            "position": "absolute",
            "top": 22,
            "zIndex": 9,
          },
          undefined,
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        Ôèè
      </Text>
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "#f1f1f1",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 97,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 32,
              },
              Object {
                "color": "#000",
              },
            ]
          }
        >
          title here
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "column",
                  "paddingRight": 10,
                  "paddingTop": 20,
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 14,
                    "textAlign": "right",
                  },
                  Object {
                    "color": "#000",
                  },
                ]
              }
            >
              #1
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 14,
                    "textAlign": "right",
                  },
                  Object {
                    "color": "#000",
                  },
                ]
              }
            >
              lvl 
              100
            </Text>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 96,
                },
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            200
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 24,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
            Object {
              "color": "#000",
            },
          ]
        }
      >
        subtitle here
      </Text>
      <View
        style={
          Array [
            Object {
              "height": 100,
            },
            undefined,
          ]
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
